### PR TITLE
Error pages for more sections

### PR DESCRIPTION
**Pull request by gnunicorn**
- [x] Task
  - [x] TaskList List
  - [x] TaskList Details
  - [x] Task Item Details
- [x] Events
  - [x] Events List
  - [x] Event Details

With tests, on top of #2089 of course.

Additionally this refactors the `AsyncRsvpStatusNotifier` into a typed variant that uses the `RsvpStatusTag` as it should be done.

### CODE REVIEW

File: app/pubspec.yaml [hunk 1]
```diff
@@ -148,6 +148,7 @@ dependencies:
   open_filex: ^4.5.0
   phosphor_flutter: ^2.1.0
   device_calendar: ^4.3.2
+  quickalert: ^1.1.0
```
Comment (gtalha07) [inline app/pubspec.yaml:151]: Aren't we using ```EasyLoading``` package to show different data states? Why this one is needed on top??

File: app/test/features/events/error_pages_test.dart [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import 'package:acter/features/events/providers/event_providers.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import '../../helpers/error_helpers.dart';
+import '../../helpers/mock_event_providers.dart';
+import '../../helpers/mock_space_providers.dart';
+import '../../helpers/test_util.dart';
+
+void main() {
+  group('Event List Error Pages', () {
```
Comment (gtalha07) [inline app/test/features/events/error_pages_test.dart:16]: Protests! More like pro tests 😄 👏🏻

**Review by gtalha07 — APPROVED** [review]
Nice additions. Overall lgtm!

File: app/pubspec.yaml (hunk 1 above)
Comment (gnunicorn) [reply app/pubspec.yaml:151]: easyloading is transative, dialog-oriented and non-disruptive. quickalert provides further widget for more prominent, larger and more intrusive widgets. They are complementary.

Comment (gtalha07) [reply app/pubspec.yaml:151]: If that is so, do we want to consider unify all alerts be approached according to this package? If it also supports loading/error states?

**Pull request merged**